tests/test_util.py

```diff
@@ -1,22 +1,27 @@
 import attr
 import pytest
 
-from autoextract_poet.util import remove_unknown_fields
+from autoextract_poet.util import split_in_unknown_and_known_fields
 
 
-@attr.s(auto_attribs=True)
+@attr.s(auto_attribs=True, slots=True)
 class _TestItem:
     k1: int
     k2: int
 
 
-def test_remove_unknown_fields():
-    expected = dict(k1=1, k2=2)
-    prep1 = remove_unknown_fields(dict(k1=1, k2=2, extra=3), _TestItem)
-    prep2 = remove_unknown_fields(dict(k1=1, k2=2, extra=3), _TestItem)
-    assert prep1 == prep2
-    assert prep2 == expected
-    assert attr.asdict(_TestItem(**prep2)) == expected
+def test_split_in_unknown_and_known_fields():
+    input = dict(k1=1, k2=2, extra=3)
+    unknown, known = split_in_unknown_and_known_fields(input, _TestItem)
+    assert attr.asdict(_TestItem(**known)) == dict(k1=1, k2=2)
+    assert unknown == dict(extra=3)
+
+    unknown, known = split_in_unknown_and_known_fields(known, _TestItem)
+    assert unknown == {}
+
+    for empty_input in ({}, None):
+        ret = split_in_unknown_and_known_fields(empty_input, _TestItem)
+        assert ret == ({}, {})
 
     with pytest.raises(ValueError):
-        remove_unknown_fields(expected, str)
\ No newline at end of file
+        split_in_unknown_and_known_fields(input, str)
```

Comment:
I would propose to add the same check for some custom not `attr.s` class.

Reply:
What do you mean? `str` is already not `attr.s`. So maybe you think that there could be a difference if we use a custom class like:

```py
class MyClass:
  a = 2
```

But this is exactly the same as if we use `str`. Adding it would mean more code without making tests more robust.